src/commons/workspace/WorkspaceReducer.ts

```diff
@@ -784,6 +786,34 @@ const newWorkspaceReducer = createReducer(defaultWorkspaceManager, builder => {
 
       state[workspaceLocation].editorTabs = newEditorTabs;
     })
+    .addCase(updateTabReadOnly, (state, action) => {
+      const workspaceLocation = getWorkspaceLocation(action);
+      const { editorTabIndex, isReadOnly } = action.payload;
+      if (isNull(editorTabIndex) || editorTabIndex < 0) {
+        return state;
+      }
+      const editorTabs = state[workspaceLocation].editorTabs;
+      const newEditorTabs = editorTabs.map((editorTab: EditorTabState, index: number | null) =>
+        index === editorTabIndex
+          ? {
+              ...editorTab,
+              readOnly: isReadOnly
+            }
+          : editorTab
+      );
+
+      if (isEqual(editorTabs, newEditorTabs)) {
+        return state;
```

Comment:
Instead of doing this very expensive computation, just read the old state (O(1) time), and read its `readOnly` old value (also O(1) time). Then compare it with the payload. If they are equal, return the old state. If they are not equal, compute the new state and return it.